### PR TITLE
Nj 172 2024pdf

### PR DESCRIPTION
**Pull request by mluedke2**
## Link to pivotal/JIRA issue

https://github.com/newjersey/affordability-pm/issues/172

## Is PM acceptance required?
- Yes - don't merge until JIRA issue is accepted!

**Reminder**: merge main into this branch and get green tests before merging to main
## What was done?
- In the implementation, it was actually a clean swap!
- In the tests, 99% of the updates needed were just a different type of empty expectation (`nil` instead of `""` when a field is not set)
- The one weird field that I was never able to really figure out is `Group1qualwi5ab`, which is the checkbox for the spouse's year of death when QSS.
  -  I manually checked that the field actually still behaves correctly (it does, screenshots below) and ended up adjusting the expectations to what the values become. Even if they aren't what I would normally expect, there is a weird predictability to them so we can still test on them.
  
## How to test?
- All cases and values should appear the same, just on 2024 PDF now.

## Screenshots (for visual changes)
Here is the PDF still filling out correctly for the Spouse year of death:

Input is `"0"`, meaning `"TwoYearPrior"`. Output is `"Choice1"` and the below:
image

---

Input is `"1"`, meaning `"LastYear"`. Output is `"Off"` and the below:
image

---

Input is `nil`, meaning not that filing status. Output is `""` and the below:
image

### CODE REVIEW

**Review by mrotondo — APPROVED** [review]

**Comment by mmazanec22** [comment]
When I manually test with the qualifying widow/er profile I don't see either 2022 or 2023 checked. This might be a problem with the xml itself but I'm not sure

File: spec/lib/pdf_filler/nj1040_pdf_spec.rb [hunk 1]
```diff
@@ -89,15 +89,15 @@
...
-          expect(pdf_fields["undefined_7"]).to eq ""
-          expect(pdf_fields["undefined_8"]).to eq ""
-          expect(pdf_fields["Enter spousesCU partners SSN"]).to eq ""
-          expect(pdf_fields["Text31"]).to eq ""
-          expect(pdf_fields["Text32"]).to eq ""
-          expect(pdf_fields["Text33"]).to eq ""
-          expect(pdf_fields["Text34"]).to eq ""
-          expect(pdf_fields["Text35"]).to eq ""
-          expect(pdf_fields["Text36"]).to eq ""
+          expect(pdf_fields["undefined_7"]).to eq nil
```
Comment (mmazanec22) [inline spec/lib/pdf_filler/nj1040_pdf_spec.rb:92]: Trying to wrap my head around why this changed. So these are all nil by default in the new PDF, whereas before the PDF had an empty string value in the field?

File: spec/lib/pdf_filler/nj1040_pdf_spec.rb [hunk 2]
```diff
@@ -989,6 +989,20 @@
...
+            "Choice1"
+          when "1"
+            "Off"
+          when nil
+            ""
+          else
+            input
+          end
+        end
+
```
[the remark this answers is not in the document]
Comment (mmazanec22) [reply spec/lib/pdf_filler/nj1040_pdf_spec.rb:1005]: [pebble] why map instead of just changing the expected value?

**Review by aloverso — APPROVED** [review]

File: spec/lib/pdf_filler/nj1040_pdf_spec.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (aloverso) [reply spec/lib/pdf_filler/nj1040_pdf_spec.rb:1005]: +1

[the remark this answers is not in the document]
Comment (mluedke2) [reply spec/lib/pdf_filler/nj1040_pdf_spec.rb:1005]: @mmazanec22 i wanted to have the map be part of the explanation -- i know it has a comment anyway, but the comment i'd need to attach if just changing the expectation felt convoluted

File: spec/lib/pdf_filler/nj1040_pdf_spec.rb (hunk 1 above)
Comment (mluedke2) [reply spec/lib/pdf_filler/nj1040_pdf_spec.rb:92]: the best clue I have is the output from `pdftk`. compare the first two fields

from 2023:
```
---
FieldType: Text
FieldName: undefined
FieldFlags: 12582912
FieldValue:
FieldJustification: Center
FieldMaxLength: 1
---
```

and from 2024:
```
---
FieldType: Text
FieldName: undefined
FieldFlags: 12582912
FieldValue: 
FieldJustification: Center
FieldMaxLength: 1
---
```

identical **except** there is an extra space in the `FieldValue` of the 2024 version. I think this is related to the difference in code too, though I'm not sure why.

**Comment by mluedke2** [comment]
> When I manually test with the qualifying widow/er profile I don't see either 2022 or 2023 checked. This might be a problem with the xml itself but I'm not sure

@mmazanec22 I tried this and got the same result. In fact this is an existing bug and exists on `main`. So I'll merge this and then create a follow-up bug ticket to fix that.

**Pull request merged**